joomla/joomla-cms
# [4.1] Actionlogs: Added parameter to disable relative timestamps.

Based on a request in the German support forum: https://forum.joomla.de/thread/12928-joomla-benutzeraktivit%C3%A4ten-datum-und-uhrzeit-protokollieren/

The actionlogs component displays date and time of an action relative to the current time ("X minutes/hours/days ago"). While this might be useful to some people, others prefer to see only the exact timestamp when the action happened. In J3, this timestamp is hidden in a tooltip. In J4, the display is better, as the timestamp is in small font below the relative one. But this becomes confusing and redundant again, at least for actions that happened more than 4 weeks ago. In this case, we have two times the same date in a different format:
![joomla-pr-actionlog-date-relative-old](https://user-images.githubusercontent.com/1410135/102339341-4f6aac00-3f95-11eb-918a-d5eb07207a89.png)


### Summary of Changes

As some users might be fine with the relative date format, I introduced a parameter in the configuration of com_actionlogs, that allows to disable the relative display. To not change current experience, it defaults to the current behaviour, i.e., displaying both relative and absolute date.


### Testing Instructions

1. Make sure you have the action logs plugins enabled and actions are logged.
2. Perform some actions in backend.
3. Navigate to "Users - User Actions Log".
4. Look at the list. It should look like the image at "actual result" below.
5. Apply this PR.
6. Reload the action log list. It should still look the same.
7. Click on "Options" and set the new parameter "Show Relative Date/Time" to "No".
8. Save & Close
9. Look at the action log list again.


### Actual result BEFORE applying this Pull Request

![joomla-pr-actionlog-date-relative-yes](https://user-images.githubusercontent.com/1410135/102339747-e9325900-3f95-11eb-963a-d7847d889214.png)


### Expected result AFTER applying this Pull Request

![joomla-pr-actionlog-date-relative-options](https://user-images.githubusercontent.com/1410135/102339824-07985480-3f96-11eb-9257-73e094be23b9.png)

![joomla-pr-actionlog-date-relative-no](https://user-images.githubusercontent.com/1410135/102339742-e6376880-3f95-11eb-9f26-abbe935850fa.png)


### Documentation Changes Required

Should be documented at https://docs.joomla.org/Help39:Components_User_Actionlogs_Options
